nb-engine/nb-engine-core/src/main/java/io/nosqlbench/engine/api/activityimpl/uniform/Activity.java

```diff
@@ -0,0 +1,969 @@
+        /// methods should also be supported.
+
+        SequencerType sequencerType = config.getOptional("seq").map(SequencerType::valueOf)
+            .orElse(SequencerType.bucket);
+
+        SequencePlanner<OpDispenser<? extends CycleOp<?>>> planner = new SequencePlanner<>(
+            sequencerType);
+
+        for (OpDispenser<?> dispenser : dispensers) {
+            planner.addOp(dispenser, d -> d.getRatio());
+        }
+        OpSequence<OpDispenser<? extends CycleOp<?>>> sequence = planner.resolve();
+        return sequence;
+
+        // TODO: Perhaps, op templates should be split into core/reserved partition and another, with a proxy
+        // object retained for the core elements
+
+        //endregion
+
+
+        //        Optional<String> defaultDriverName = activityDef.getParams().getOptionalString("driver");
+        //        Optional<DriverAdapter<?, ?>> defaultAdapter = activityDef.getParams()
+        //            .getOptionalString("driver")
+        //            .flatMap(name -> ServiceSelector.of(name, ServiceLoader.load(DriverAdapterLoader.class)).get())
+        //            .map(l -> l.load(this, NBLabels.forKV()));
+        //
+        //        if (defaultDriverName.isPresent() && defaultAdapter.isEmpty()) {
+        //            throw new BasicError("Unable to load '" + defaultDriverName.get() + "' driver adapter.\n" + "Rebuild NB5 to include this " + "driver adapter. Change" + " '<activeByDefault>false</activeByDefault>' for the driver in" + " './nb-adapters/pom.xml' and" + " './nb-adapters/nb-adapters-included/pom.xml' first.");
+        //        }
+
+        //        NBConfigModel supersetConfig = ConfigModel.of(Activity.class).add(yamlmodel);
+        //        Optional<String> defaultDriverOption = defaultDriverName;
+        //        ConcurrentHashMap<String, OpMapper<? extends CycleOp<?>, ? extends Space>> mappers = new ConcurrentHashMap<>();
+
+        //        List<ParsedOp> allParsedOps = loadOpTemplates(defaultAdapter.orElse(null), false, false).stream()
+        //            .map(ot -> upconvert(ot, defaultDriverOption, yamlmodel, supersetConfig, mappers, adapterlist))
+        //            .toList();
+
+        //        OpLookup lookup = new OpLookupService(() -> allParsedOps);
+
+        //        TagFilter ts = new TagFilter(activityDef.getParams().getOptionalString("tags").orElse(""));
+        //        List<ParsedOp> activeParsedOps = ts.filter(allParsedOps);
+
+        //        if (defaultDriverOption.isPresent()) {
+        //            long matchingDefault = mappers.keySet().stream().filter(n -> n.equals(defaultDriverOption.get())).count();
+        //            if (0 == matchingDefault) {
+        //                logger.warn(
+        //                    "All op templates used a different driver than the default '{}'",
+        //                    defaultDriverOption.get()
+        //                );
+        //            }
```

Comment:
In the new object model it's unclear to me how to make this test. Perhaps by looking into expanding how `adapterCache` map functions.

Reply:
Yes, the adapter cache is a product of op resolution, and can be checked for canonical mapping of active adapters. We can make sure this is exposed. We might want to do this where we isolate a particular verification checkpoint where we assert sanity on the assembled op dispensers.

Reply:
adapterCache includes the default Adaptor. Instead scan the opTemplates into an adapter list and check vs. the default adapter.

Reply:
adapterCache should not include the default unless it was referenced by an op template. Also, I'm beginning to think this is not really an issue, since there are plenty of cases where a default adapter might be supplied and not used (test automation)

Reply:
I'd like to take this scope as a follow-on for this PR if that's ok.